### ISSUE TITLE
Property-Change is not propagated when using java record as @ConfigationProperties class

### ISSUE THREAD

**Issue by tschniewind** [spring-cloud/spring-cloud-kubernetes#1268]
Spring Boot Version: 2.7.9
Spring Cloud Kubernetes Version: 2.1.6
Maven dependency:

```
<dependency>
    <groupId>org.springframework.cloud</groupId>
    <artifactId>spring-cloud-starter-kubernetes-client-config</artifactId>
</dependency>
```

I am using a java record to hold my configuration properties:
```
@ConfigurationProperties(prefix = "my-service.config")
public record ApplicationConfig(boolean processIncomingMessages) {}
```

When i change my confi-map the change itself is detected, but the ApplicationConfig-Bean is not replaced so the change can't take place within my application.

When i use a plan java object, with setter and getter everything works fine:
```
@ConfigurationProperties(prefix = "my-service.config")
public class ApplicationConfig {
    private boolean processIncomingMessages;
    void setProcessIncomingMessages(final boolean processIncomingMessages) {
        this.processIncomingMessages = processIncomingMessages;
    }
    public boolean processIncomingMessages() {
        return this.processIncomingMessages;
    }
}
```

**Comment by wind57** [comment]
I remember something similar a while back being reported for a kotlin based project, where an immutable object had the same exact problem. I'll dig to see if this is something known or has to do with spring-boot.

**Comment by ryanjbaxter** [comment]
I am going to guess if this is a problem its not specific to Spring Cloud Kubernetes and that this would likely be able to be reproduced with a Spring Boot app with Spring Cloud Commons and hitting the `/refresh` actuator endpoint

**Comment by ryanjbaxter** [comment]
So even without Spring Cloud Kubernetes on the classpath you can reproduce the problem?

**Comment by wind57** [comment]
this is what I mean when I said that it is "immutable" - a `record` is such, so it can not be refreshed, see [this explanation and reasonings](https://github.com/spring-cloud/spring-cloud-config/issues/1547).

There is not much we can do from our project and it has nothing to do with it :)

**Comment by tschniewind** [comment]
Yes, I agree with you. It is a dilemma that could not be resolved by code, immutable should stay immutable. 
Many thanks for your support!
